### PR TITLE
maintainers: Split up NXP maintainer group

### PR DESCRIPTION
**Pull request by dleach02**
Split the NXP maintainer group into NXP S32, NXP Xtensa, and NXP.

### CODE REVIEW

**Comment by dleach02** [comment]
@stephanosio, @nashif, not exactly sure how to test maintainer file changes to ensure the bots process it correctly.

**Review by nashif — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by dleach02** [comment]
We are having an internal discussion about one part of this so please hold off on merging

**Review by manuargue — CHANGES_REQUESTED** [review]

**Review by dbaluta — DISMISSED** [review]

**Review by manuargue — DISMISSED** [review]

File: MAINTAINERS.yml
Comment (dleach02) [reply MAINTAINERS.yml:3182]: was wondering what the problem was... thanks

**Review by manuargue — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: MAINTAINERS.yml
Comment (fabiobaltieri) [inline MAINTAINERS.yml:3194]: should this have a qualifier? MCUX or something

**Review by fabiobaltieri — APPROVED** [review]

File: MAINTAINERS.yml
Comment (dleach02) [reply MAINTAINERS.yml:3194]: No. While some of this is MCUX there are others like "LPC" and soon will be another. So we don't want to tie a particular branding to the larger bucket at this time.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**
